digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/MassTransitStopService.scala

```diff
@@ -85,15 +86,76 @@ trait MassTransitStopService extends PointAssetOperations {
           left join single_choice_value s on s.asset_id = a.id and s.property_id = p.id and p.property_type = 'single_choice'
           left join text_property_value tp on tp.asset_id = a.id and tp.property_id = p.id and (p.property_type = 'text' or p.property_type = 'long_text')
           left join multiple_choice_value mc on mc.asset_id = a.id and mc.property_id = p.id and p.property_type = 'multiple_choice'
+          left join number_property_value np on np.asset_id = a.id and np.property_id = p.id and p.property_type = 'read_only_number'
           left join enumerated_value e on mc.enumerated_value_id = e.id or s.enumerated_value_id = e.id
       """
     queryToPersistedMassTransitStops(queryFilter(query))
   }
 
+  override def updateFloating(id: Long, floating: Boolean, floatingReason: Option[FloatingReason]) = {
+    super.updateFloating(id, floating, floatingReason)
+
+    floatingReason match {
+      case None =>
+        deleteFloatingReasonValue(id)
+      case Some(reason) =>
+        updateFloatingReasonValue(id, reason)
+    }
+  }
+
+  override def isFloating(persistedAsset: PersistedPointAsset, roadLinkOption: Option[VVHRoadlink]): (Boolean, Option[FloatingReason]) = {
+
+    roadLinkOption match {
+      case None => return super.isFloating(persistedAsset, roadLinkOption)
+      case Some(roadLink) =>
+        val administrationClass = massTransitStopDao.getAssetAdministrationClass(persistedAsset.id)
```

Comment:
Confusing to have both administrationClass and administrativeClass. Could they both be administrativeClass?